src/jit/compiler.cpp

```diff
@@ -2620,7 +2583,9 @@ void Compiler::compSetProcessor()
             }
             if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_AVX2))
             {
-                if (configEnableISA(InstructionSet_AVX2))
+                // COMPlus_EnableAVX is also used to control the code generation of
+                // System.Numerics.Vectors and floating-point arithmetics
+                if (configEnableISA(InstructionSet_AVX) && configEnableISA(InstructionSet_AVX2))
```

Comment:
> Where is this being handled now? This was the condition that caused us not to generate AVX code during crossgen, as we can't be assured that the target will be the same.

@CarolEidt I am using `InstructionSet_AVX` and `InstructionSet_AVX2` instead of `UseAVX`, which is already guarded by `!jitFlags.IsSet(JitFlags::JIT_FLAG_PREJIT)`.